### PR TITLE
Sample stable inputs in tests of group normalization

### LINKED ISSUE #6911: `test_group_normalization` is flaky
This is part of https://github.com/chainer/chainer/issues/6903 effort.

- Failure example: https://ci.preferred.jp/chainer.py27and35/2138/
- Failure test case: `GroupNormalizationTest_param_8_{dtype=float32, shape=(3, 8), groups=4}.test_backward_gpu_without_cudnn`
- Tolerance:
```
E   Not equal to tolerance rtol=0.01, atol=0.001
E   
E   Mismatch: 100%
E   Max absolute difference: 0.15960103
E   Max relative difference: 0.0476208
```

### PR DESCRIPTION
**Pull request by toslunar**
Fix #6911.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/normalization_tests/test_group_normalization.py [hunk 1]
```diff
@@ -63,9 +63,21 @@ def setUp(self):
...
+        # sample x such that x.std >= min_std
+        min_std = 0.2 if self.dtype == numpy.float16 else 0.02
+        retry = 0
+        while True:
+            x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
+            x_groups = x.reshape(shape[0], self.groups, -1)
+            if x_groups.std(axis=2).min() >= min_std:
+                break
+            retry += 1
+            assert retry <= 10, 'Too many retries to generate inputs'
```
Comment (emcastillo) [inline tests/chainer_tests/functions_tests/normalization_tests/test_group_normalization.py]: I find this fine, I am just curious about how many retries does it take on average to sample a good input :).

Thanks!

Comment (toslunar) [reply tests/chainer_tests/functions_tests/normalization_tests/test_group_normalization.py]: For example, `shape=(3, 20), groups=4` samples 5 per group.  For each group std ≥ 0.2 with probability ~97.8%.
```
>>> (np.random.uniform(-1, 1, (1000000, 5)).std(axis=1) > 0.2).sum()
978456
```
A try satisfies the condition for all groups with probability at least `0.97 ** (3 * 4)`.  Thus it fails 10 tries with probability
```
>>> (1 - 0.97 ** (3 * 4)) ** 10
7.235207665894931e-06
```

... I see.  It's not sufficiently small.

Comment (emcastillo) [reply tests/chainer_tests/functions_tests/normalization_tests/test_group_normalization.py]: Thanks a lot for the great explanation!

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
CI unrelated

**Pull request merged**